### PR TITLE
Feature/buttons

### PR DESCRIPTION
**Pull request by William-Blackie**
Pattern library link; http://0.0.0.0:3000/pattern-library/pattern/patterns/components/buttons/buttons.html

Video demo;
https://user-images.githubusercontent.com/22497837/180213009-e861cba2-c809-43e9-9c5d-bf3ca8456211.mp4

I've added the button styles, including a basic animation for the arrow/hover effect. There might be others I haven't seen, please shout if there are. 

I've commented on the diff where I'm unsure of my approach.

### CODE REVIEW

File: wagtail_dot_org/static/sass/abstracts/_variables.scss [hunk 1]
```diff
@@ -3,8 +3,10 @@ $color--black: #000;
 $color--off-black: #262626;
 $color--indigo: #2E1F5E;
 $color--teal: #007D7E;
+$color--teal-100: #80D7D8;
```
Comment (William-Blackie) [inline wagtail_dot_org/static/sass/abstracts/_variables.scss:6]: Not sure if this is the best naming convention, but it's still teal...

File: wagtail_dot_org/static/sass/atoms/icons.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.arrow {
+    fill: $color--teal-100;
+    height: 19px;
+    width: 25px;
+}
```
Comment (William-Blackie) [inline wagtail_dot_org/static/sass/atoms/icons.scss:5]: Does a more generic icon styles file make sense?

File: wagtail_dot_org/static/sass/components/_buttons.scss [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    border-radius: 6px;
+    border: 1px solid $color--off-black;
+    margin-left: auto;
+    margin-right: auto;
+
+    padding: 15px 30px;
+    color: $color--off-black;
+    
+    &:hover{
```
Comment (William-Blackie) [inline wagtail_dot_org/static/sass/components/_buttons.scss:14]: Originally, I wanted to put this in the icon styles, but it makes more sense to put this here to be able to reach the parent class.

It does mean there's a fair amount of re-writting of transforms, is there a better way to do this?

**Comment by chris-lawton** [comment]
Discussed points above in person. Thanks Will!

**Pull request merged**